dags/utils/config_parser.py

```diff
@@ -29,13 +29,25 @@ def __init__(self, path=None):
     def _get_country(self, country) -> FallbackDict:
         return FallbackDict(self.raw_config, self.country_config[self._country_lower(country)])
 
-    def name_output_file_generic(self, geo_extent, category, theme, geometry, scale, source, permission,
-                                 free_text=None):
+    def _get_name_output_file_generic(self, country: str, filename_field: FallbackDict) -> str:
+        geo_extent = self.get_iso3(country).lower()
+        file_name = self._name_output_file_generic(geo_extent, filename_field['category'], filename_field['theme'],
+                                                   filename_field['geometry'], filename_field['scale'],
+                                                   filename_field['source'], filename_field['suffix'])
+        return file_name
+
+    def _name_output_file_generic(self, geo_extent: str, category: str, theme: str, geometry: str, scale: str,
+                                  source: str, suffix: str, permission: str = 'pp', free_text: str = None) -> str:
         file_name = f"{geo_extent}_{category}_{theme}_{geometry}_{scale}_{source}_{permission}"
-        if free_text:
+        if free_text is not None:
             file_name += f"_{free_text}"
+        file_name += suffix
```

Comment:
The dot before the file extension is lost here, this should be something like:
`file_name += f".{suffix}"`

Reply:
Oops, good catch, I will fix this!